src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Formatters/CollectionFormatter.cs

```diff
@@ -535,6 +535,116 @@ protected override Queue<T> Complete(Queue<T> intermediateCollection)
         }
     }
 
+#if NET6_0_OR_GREATER
+    public sealed class PriorityQueueFormatter<TElement, TPriority> : IMessagePackFormatter<PriorityQueue<TElement, TPriority>?>
+    {
+        private readonly IComparer<TPriority>? comparer;
+
+        public PriorityQueueFormatter()
+        {
+            comparer = default;
+        }
+
+        public PriorityQueueFormatter(IComparer<TPriority>? comparer)
+        {
+            this.comparer = comparer;
+        }
+
+        public void Serialize(ref MessagePackWriter writer, PriorityQueue<TElement, TPriority>? value, MessagePackSerializerOptions options)
+        {
+            if (value is null)
+            {
+                writer.WriteNil();
+                return;
+            }
+
+            // https://github.com/dotnet/runtime/blob/a6bf4e4b94db9c1e28f50393e0a1943892e34bed/src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs#L172
+            // Count property is reading just private field _size;
+            var count = value.Count;
+            writer.WriteMapHeader(count);
```

Comment:
I don't think a queue should be expressed as a map. The elements may not be unique, and maps without typing info may deserialize as a dictionary that would throw when non-unique keys are encountered.
Rather, I think we should use an array, where each array element is itself an array of two elements.

Reply:
Ah, the fact maps without type-info can be deserialized into dictionary was certainly missing from my consideration.
I changed here to use Write|ReadArrayHeader and used `ValueTupleFormatter` for each pair.